### PR TITLE
[API][Shipments] Browsing shipments

### PR DESCRIPTION
**Pull request by AdamKasp**
| Q               | A
| --------------- | -----
| Branch?         | API
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | replace #11208
| License         | MIT

### CODE REVIEW

File: src/Sylius/Behat/Context/Api/Admin/ManagingShipmentsContext.php [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        string $orderNumber,
+        string $shippingState,
+        CustomerInterface $customer,
+        ChannelInterface $channel = null
+    ): void {
+        $this->client->index();
+        
+        foreach ($this->responseChecker->getCollectionItemsWithValue($this->client->getLastResponse(), 'state', StringInflector::nameToLowercaseCode($shippingState)) as $shipment) {
+            $orderIri = $shipment['order'];
+            $this->client->showByIri($orderIri);
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Api/Admin/ManagingShipmentsContext.php]: We can take response directly for this call (https://github.com/Sylius/Sylius/pull/11220/files#diff-7cd67ba0c744465221bda1283b08928dR22)

File: src/Sylius/Behat/Client/ResponseChecker.php [hunk 2]
```diff
@@ -100,6 +100,17 @@ public function hasItemWithTranslation(Response $response, string $locale, strin
         return false;
     }
 
+    public function hasItemWithKeysAndValues(Response $response, array $parameters): bool
```
Comment (GSadee) [inline src/Sylius/Behat/Client/ResponseChecker.php]:
```suggestion
    public function hasItemWithValues(Response $response, array $parameters): bool
```

File: src/Sylius/Behat/Client/ResponseChecker.php [hunk 3]
```diff
@@ -108,4 +119,14 @@ private function getResponseContentValue(Response $response, string $key)
 
         return $content[$key];
     }
+
+    private function elementHasKeysAndValues(array $element, array $parameters): bool {
```
Comment (GSadee) [inline src/Sylius/Behat/Client/ResponseChecker.php]:
```suggestion
    private function itemHasValues(array $element, array $parameters): bool {
```

File: src/Sylius/Behat/Context/Api/Admin/ManagingShipmentsContext.php [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+     * @Then /^I should see shipment for the ("[^"]+" order) as (\d+)(?:|st|nd|rd|th) in the list$/
+     */
+    public function iShouldSeeShipmentForTheOrderInTheList(OrderInterface $order, int $position): void
+    {
+        Assert::true(
+            $this->responseChecker->hasItemOnPositionWithValue(
+                $this->client->getLastResponse(),
+                --$position,
+                'order',
+                $this->iriConverter->getIriFromItem($order)),
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Admin/ManagingShipmentsContext.php]:
```suggestion
                $this->iriConverter->getIriFromItem($order)
            ),
```

File: src/Sylius/Behat/Context/Api/Admin/ManagingShipmentsContext.php [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+     */
+    public function shipmentOfOrderShouldBe(
+        string $orderNumber,
+        string $shippingState,
+        CustomerInterface $customer,
+        ChannelInterface $channel = null
+    ): void {
+        $this->client->index();
+
+        foreach ($this->responseChecker->getCollectionItemsWithValue($this->client->getLastResponse(), 'state', StringInflector::nameToLowercaseCode($shippingState)) as $shipment) {
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Admin/ManagingShipmentsContext.php]:
```suggestion
        foreach ($this->responseChecker->getCollectionItemsWithValue($this->client->index(), 'state', StringInflector::nameToLowercaseCode($shippingState)) as $shipment) {
```

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Channel.xml [hunk 6]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" ?>
+
+<resources xmlns="https://api-platform.com/schema/metadata"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="https://api-platform.com/schema/metadata https://api-platform.com/schema/metadata/metadata-2.0.xsd"
+>
+    <resource
+        class="%sylius.model.channel.class%"
+        shortName="Channel"
+    >
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Channel.xml]:
```suggestion
    <resource class="%sylius.model.channel.class%" shortName="Channel">
```

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Channel.xml [hunk 7]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" ?>
+
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Channel.xml]: Add license docblock

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Shipment.xml [hunk 8]
```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0" ?>
+
+<resources xmlns="https://api-platform.com/schema/metadata"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="https://api-platform.com/schema/metadata https://api-platform.com/schema/metadata/metadata-2.0.xsd"
+>
+    <resource
+        class="%sylius.model.shipment.class%"
+        shortName="Shipment"
+    >
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Shipment.xml]: Could be in one line here and in other xml files

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Shipment.xml [hunk 9]
```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0" ?>
+
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Shipment.xml]: Add license docblock here and in other xml files

File: src/Sylius/Bundle/ApiBundle/Resources/config/serializer/Channel.xml [hunk 10]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" ?>
+
+<serializer xmlns="http://symfony.com/schema/dic/serializer-mapping"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/serializer-mapping https://symfony.com/schema/dic/serializer-mapping/serializer-mapping-1.0.xsd">
+
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Resources/config/serializer/Channel.xml]:
```suggestion
            xsi:schemaLocation="http://symfony.com/schema/dic/serializer-mapping https://symfony.com/schema/dic/serializer-mapping/serializer-mapping-1.0.xsd"
>
```

**Comment by GSadee** [comment]
Thanks, Adam! :tada:

**Comment by lchrusciel** [comment]
Part of #11250

**Pull request merged**